lib/slack.ml

```diff
@@ -281,13 +272,41 @@ let generate_status_notification (cfg : Config_t.config) (notification : status_
       ]
   in
   let summary =
+    let state_info =
+      match state with
+      | Success -> "succeeded"
+      | Failure -> "failed"
+      | Error -> "error"
+      | _ ->
+        invalid_arg
+          (sprintf "Monorobot doesn't know how to generate notification for the unexpected event %s of %s"
+             (string_of_status_state state) sha
+          )
+    in
+    let commit_message = first_line message in
     match target_url with
-    | None ->
-      sprintf "<%s|[%s]> CI Build Status notification: %s" repository.url repository.full_name state_info
-      (* in case the CI run is not using buildkite *)
+    | None -> sprintf "<%s|[%s]> CI Build Status notification: %s" repository.url repository.full_name state_info
     | Some target_url ->
-      sprintf "<%s|[%s]> CI Build Status notification for <%s|%s>: %s" repository.url repository.full_name target_url
-        context state_info
+      let default_summary =
+        sprintf "<%s|[%s]> CI Build Status notification for <%s|%s>: %s" repository.url repository.full_name target_url
+          context state_info
+      in
+      if not is_buildkite then default_summary
+      else (
+        (* Keep only the portion of the url before /builds/... *)
+        let pipeline_url =
+          let rec loop = function
+            | [] -> None
+            | "builds" :: l -> Some (List.rev l |> String.concat ~sep:"/")
+            | _ :: l -> loop l
+          in
+          let url_parts = target_url |> String.split ~on:'/' |> List.rev in
+          loop url_parts
```

Comment:
```
# #require "extlib";;
# ExtString.String.split;;
- : string -> string -> string * string = <fun>
# ExtString.String.split "https://buildkite.com/foo/builds/10120" "builds";;
- : string * string = ("https://buildkite.com/foo/", "/10120")
```

Reply:
I wanted to handle the case where the name of the `pipeline` contains `builds`:

```ocaml
utop # ExtLib.String.split "https://buildkite.com/builds-monorepo/builds/10120" "builds";;
- string * string = ("https://buildkite.com/", "-monorepo/builds/10120")
```

But maybe I'm overthinking this, I'll switch to the simpler version

Reply:
Ah ok. If you want to do that I’d use pattern matching to simplify things

```ocaml
match String.split ~on:'/' target_url with
| "https:" :: "" :: "buildkite.com" :: pipeline :: "builds" :: build_id -> …
| _ -> failwith "invalid url"
```